project_issue_task_sync/models/project_task.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    sync_tasks_issues = fields.Boolean(
+        related=['project_id', 'sync_tasks_issues']
+    )
+    issue_ids = fields.One2many('project.issue', 'task_id', 'Issues')
+
+    def get_or_create_binded_issue(self):
+        if self.issue_ids:
+            return self.issue_ids[0]
+        issue = self.env['project.issue'].with_context(
+            mail_notrack=True, is_sync_operation=True).create({
+                'project_id':  self.project_id.id,
+                'name': self.name,
+                'task_id':  self.id,
+                'user_id': self.user_id.id,
+                'stage_id': self.stage_id.id,
+                'description': self.description or 'No Description'
+                })
+        return issue
+
+    @api.multi
+    def set_binded_issue_vals(self):
+        for this in self:
+            if not this.env.context.get('is_sync_operation'):
+                if not this.issue_ids:
+                    issue = this.get_or_create_binded_issue()
+                for issue in this.issue_ids:
+                    vals = self.env['project.issue'].get_changed_vals(
+                        this, issue
+                    )
+                    issue.with_context(
+                        mail_notrack=True, is_sync_operation=True
+                    ).write(vals)
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, vals):
+        this = super(ProjectTask, self).create(vals)
+        if this.project_id.sync_tasks_issues:
+            this.get_or_create_binded_issue()
+        return this
```

Comment:
Does it make sense to have a two way automatic creation?
I can see the point to automatically create a Task for each new Issue, but I will probably prefer to not automatically create an Issue for new Tasks. That should be an additional flag on the Project, or perhaps a button/"More" option on Tasks.
Comments?